crates/cext/src/transpiler/passes/optimize_1q_decomposition.rs

```diff
@@ -0,0 +1,103 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use crate::pointers::{const_ptr_as_ref, mut_ptr_as_ref};
+use qiskit_circuit::{
+    circuit_data::CircuitData, converters::dag_to_circuit, dag_circuit::DAGCircuit,
+};
+use qiskit_transpiler::{passes::run_optimize_1q_gates_decomposition, target::Target};
+
+/// @ingroup QkTranspilerPasses
+/// Runs the Optimize1qGatesDecomposition pass in standalone mode on a circuit.
+///
+/// Optimize1qGatesDecomposition, as its name implies, optimizes chains of single-qubit
+/// gates by combining them into a single gate.
+///
+/// The decision of whether to replace the original chain depends on:
+/// - If the original chain was out of basis.
+/// - If the original chain was in basis but the replacement has lower error rates.
+/// - If the original chain is an identity (chain gets removed).
+///
+/// The error is the combined multiplication of the errors of individual gates on the
+/// qubit it operates on.
+///
+/// @param circuit A pointer to the ``QkCircuit`` object to transform.
+/// @param target A pointer to the ``QkTarget`` object or a null pointer.
+/// In the case a null pointer is provided the pass calculates the error
+/// of each instruction based on each one-qubit sequence's length, and
+/// will support all basis gates on its Euler basis set.
```

Comment:
I'm not sure I fully understand the current formulation: if we don't have the target we can't calculate any errors right? Is the following suggestion maybe clearer (assuming that it is correct what I wrote 😄)?
```suggestion
/// In the case a null pointer is provided and gate errors are unknown
/// the pass will choose the sequence with the least amount of gates,
/// and will support all basis gates on its Euler basis set.
```

Reply:
Added to 19f55a61197d9d6ac5b188778e4a2dd98b020b1c